### PR TITLE
Hot reload improvements

### PR DESCRIPTION
**Pull request by Hixie**
- show the next error fully after a hot reload
- hide _AssertionError in stacks
- immediately rebuild after a reassemble, so that hit tests work
- catch errors when notifying global key listeners

### CODE REVIEW

File: packages/flutter/lib/src/widgets/focus.dart [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 import 'dart:async';
 
+import 'package:flutter/foundation.dart';
```
Comment (abarth) [inline packages/flutter/lib/src/widgets/focus.dart]: Is this import used for anything?

**Comment by abarth** [comment]
LGTM

File: packages/flutter/lib/src/widgets/focus.dart (hunk 1 above)
Comment (Hixie) [inline packages/flutter/lib/src/widgets/focus.dart]: huh, why didn't the analyzer notice this?

Comment (abarth) [inline packages/flutter/lib/src/widgets/focus.dart]: Probably you do reference something in there that you get from another import.

**Pull request merged**